src/ModelContextProtocol/Shared/McpSession.cs

```diff
@@ -487,77 +500,127 @@ public async Task SendMessageAsync(IJsonRpcMessage message, CancellationToken ca
         }
     }
 
-    private string CreateActivityName(string method) =>
-        $"mcp.{(_isServer ? "server" : "client")}.{_transportKind}/{method}";
+    private string CreateActivityName(string method) => method;
 
     private static string GetMethodName(IJsonRpcMessage message) =>
         message switch
         {
             JsonRpcRequest request => request.Method,
             JsonRpcNotification notification => notification.Method,
-            _ => "unknownMethod",
+            _ => "unknownMethod"
         };
 
-    private void AddStandardTags(ref TagList tags, string method)
+    private void AddTags(ref TagList tags, Activity? activity, IJsonRpcMessage message, string method)
     {
-        tags.Add("session.id", _id);
-        tags.Add("rpc.system", "jsonrpc");
-        tags.Add("rpc.jsonrpc.version", "2.0");
-        tags.Add("rpc.method", method);
+        tags.Add("mcp.method.name", method);
         tags.Add("network.transport", _transportKind);
 
-        // RPC spans convention also includes:
-        // server.address, server.port, client.address, client.port, network.peer.address, network.peer.port, network.type
-    }
-
-    private static void AddRpcRequestTags(ref TagList tags, Activity? activity, JsonRpcRequest request)
-    {
-        tags.Add("rpc.jsonrpc.request_id", request.Id.ToString());
+        // MCP convention also includes: server.address, server.port, client.address, client.port
```

Comment:
Does it?

Reply:
I updated them and this comment to be more descriptive